### DISCUSSION TITLE
Usage of querying contains `~` and Logical separator `{, }`

### DISCUSSION THREAD

**Discussion by hrdwdmrbl** [codatio/codat-docs#853] [Q&A] [answered]
Would you use it something like:

```
const codatOrderIds = ["abc123", "def456"];

const query = `transactionSourceRef.id~{${codatOrderIds.join(',')}}`;
```

to produce "transactionSourceRef.id~{abc123,def456}}"

OR, do I have to use `||`: "{transactionSourceRef.id=abc123 || transactionSourceRef.id=def456}" ?

**Comment by dcoplowe** [comment]
Hi @hrdwdmrbl,

You'll need to use `{transactionSourceRef.id=abc123 || transactionSourceRef.id=def456}` to make this query. It doesn't look like we've captured this in our docs so I'll make sure to update them accordingly.

Please let me know if you have any further questions.

Many thanks,

David

**Reply by hrdwdmrbl** [reply]
@dcoplowe  good to know, thanks. What is proper usage of `~` and `{, }` ?

**Reply by hrdwdmrbl** [reply]
@dcoplowe Could you please document the proper usage of those operators? There's absolutely nothing in your docs

**Reply by dcoplowe** [reply, marked as answer]
Hi @hrdwdmrbl,

Apologies for the delay here.

Correct usage of both the contains and the logical separator should be:
```
query={schemaProperty ~ some-string || schemaProperty ~ some-other-string}
```

Let me know how you get on here.

If you have any further questions I'd recommend reaching out to our support team. They will be able to assist you with similar questions you may have.

Many thanks,

David

**Reply by hrdwdmrbl** [reply]
Awesome! Thanks, @dcoplowe 

- I see now that `~` is substring matching.
- I also see that `{}` is for grouping, and that `&&` and `||` must (?) always be used in combination with `{}`. They are _always_ (?) used together.

One final thing that I don't understand is the use of `,` within `{}`. But sure, I will ask the support team, thanks again.

**Reply by dcoplowe** [reply]
Hi @hrdwdmrbl,

You're pushing to the limits of my knowledge here! I don't want to give you an incorrect answer so I'd recommend speaking to support about this. 

FYI we do have a ticket out to update our docs but given our current backlog it is unlikely that we will be able to improve it until Q1 '24.

Many thanks,

David
